src/imports.rs

```diff
@@ -557,20 +557,9 @@ impl UseTree {
                 self.path = vec![];
                 return self;
             }
-            UseSegmentKind::Slf(None) if self.path.is_empty() && self.visibility.is_some() => {
-                self.path = vec![];
-                return self;
-            }
```

Comment:
I take it this is where we get rid of `use self;`? I don't think there's any harm in keeping this behavior as long as we handle `use self::self` in a way that's idempotent.